master/internal/rm/kubernetesrm/resource_pool.go

```diff
@@ -190,6 +192,7 @@ func (k *kubernetesResourcePool) receiveRequestMsg(ctx *actor.Context) error {
 			delete(k.jobIDtoAddr, jobID)
 			delete(k.groupActorToID, msg.Ref)
 			delete(k.IDToGroupActor, jobID)
+			delete(k.allocRefToRunningPods, k.jobIDtoAddr[jobID])
```

Comment:
```k.jobIDtoAddr[jobID]``` will always return ```""``` since we just deleted the key above

I think it makes more sense to delete this when we remove the allocation from the task list.

Reply:
See new changes